### PR TITLE
[#2176] Set up documentation build with Sphinx

### PR DESCRIPTION
**Pull request by Bartvaderkin**
- Config and CI workflow inspired by open-forms
- Kept current old docs for now but fixed main issues to it completes CI

```shell
cd ./docs
make html
open ./_build/html/index.html
```

Taiga: [#2443](https://taiga.maykinmedia.nl/project/open-inwoner/task/2443), [#2445](https://taiga.maykinmedia.nl/project/open-inwoner/task/2445), [#2444](https://taiga.maykinmedia.nl/project/open-inwoner/task/2444). [#2446](https://taiga.maykinmedia.nl/project/open-inwoner/task/2446)

### CODE REVIEW

**Review by stevenbal — CHANGES_REQUESTED** [review]
Looks good overall, couple of comments/questions

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -148,6 +148,50 @@ jobs:
...
+          --name postgres
+
+      elasticsearch:
+        image: docker.elastic.co/elasticsearch/elasticsearch:7.9.2
+        env:
+          discovery.type: single-node
+          ES_JAVA_OPTS: '-Xms512m -Xmx512m'
+        ports:
+          - 9200:9200
+          - 9300:9300
```
[the remark this answers is not in the document]
Comment (Bartvaderkin) [reply .github/workflows/ci.yml:174]: Technically not, but the builder setups Django and then starts generating logs etc.

I'll see if we can get rid of that, would be handy.

File: docs/coding_style/index.rst [hunk 2]
```diff
@@ -6,6 +6,13 @@ Below you can find some best practices to maintain a good coding style. There
 are detailed coding style guides for the `frontend <coding_style_frontend>` and
 the `backend <coding_style_backend>`.
```
[the remark this answers is not in the document]
Comment (Bartvaderkin) [reply docs/coding_style/index.rst:7]: We'll probably remove these but good catch.

File: .readthedocs.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  apt_packages:
+    - libxml2-dev
+    - libxmlsec1-dev
+    - libxmlsec1-openssl
+    - wget
+  tools:
+    python: '3.10'
+#  jobs:
+#    post_checkout:
+#      - bash bin/rtd_post_checkout.sh
```
[the remark this answers is not in the document]
Comment (Bartvaderkin) [reply .readthedocs.yaml:21]: Maybe, but I'll remove it and we can re-steal it if we need it.

**Review by pi-sigma — APPROVED** [review]

**Pull request merged**